### DISCUSSION TITLE
Running a python script on bird detection

### DISCUSSION THREAD

**Discussion by weansdbeans** [mcguirepr89/BirdNET-Pi#1051] [Q&A]
Hello there,
I'm trying to create an "automatic birdbath" by using a USB powered pump to activate via the GPIO pins of the raspberry pi.
It's taken a bit of faffing around, but I've finally got it to the point where I have a relay module cut into the USB device and it can be activated via the GPIO. Script behaves as expected, and pump is working fine.

However, when I was originally researching this I was under the impression that I could set up an apprise notification to run the python script - this would make it easy to change script name / avoid any complications when BirdNet is updated. Except now I can't work out how to set it up. Was this ever a thing? I may have misread what apprise can actually do.
Any ideas on the simplest way of triggering a python script after a successful bird detection?

[activate_pump.txt](https://github.com/mcguirepr89/BirdNET-Pi/files/12908722/activate_pump.txt)

**Comment by ehpersonal38** [comment]
Here are some ideas- 
1. Inside your activate_pump script, have a loop that checks to see if `~/BirdNET-Pi/scripts/birds.db` has had any new entries added to the database since the check last ran. This would indicate a new bird being detected.
2. Add a script call in `server.py` at the point where it parses a new detection and writes it to the database. I would do it right after line 464: https://github.com/mcguirepr89/BirdNET-Pi/blob/main/scripts/server.py#L464.
